source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/events.py

```diff
@@ -846,22 +871,29 @@ def reset_joints_by_offset(
     asset: Articulation = env.scene[asset_cfg.name]
 
     # get default joint state
-    joint_pos = asset.data.default_joint_pos[env_ids].clone()
-    joint_vel = asset.data.default_joint_vel[env_ids].clone()
+    joint_pos_all = asset.data.joint_pos[env_ids].clone()
+    joint_vel_all = asset.data.joint_vel[env_ids].clone()
+
+    # isolate relevant joints
+    joint_pos = joint_pos_all[:, asset_cfg.joint_ids]
+    joint_vel = joint_vel_all[:, asset_cfg.joint_ids]
 
     # bias these values randomly
     joint_pos += math_utils.sample_uniform(*position_range, joint_pos.shape, joint_pos.device)
     joint_vel += math_utils.sample_uniform(*velocity_range, joint_vel.shape, joint_vel.device)
 
+    joint_pos_all[:, asset_cfg.joint_ids] = joint_pos
+    joint_vel_all[:, asset_cfg.joint_ids] = joint_vel
+
     # clamp joint pos to limits
     joint_pos_limits = asset.data.soft_joint_pos_limits[env_ids]
-    joint_pos = joint_pos.clamp_(joint_pos_limits[..., 0], joint_pos_limits[..., 1])
+    joint_pos_all = joint_pos_all.clamp_(joint_pos_limits[..., 0], joint_pos_limits[..., 1])
     # clamp joint vel to limits
     joint_vel_limits = asset.data.soft_joint_vel_limits[env_ids]
-    joint_vel = joint_vel.clamp_(-joint_vel_limits, joint_vel_limits)
+    joint_vel_all = joint_vel_all.clamp_(-joint_vel_limits, joint_vel_limits)
 
     # set into the physics simulation
-    asset.write_joint_state_to_sim(joint_pos, joint_vel, env_ids=env_ids)
+    asset.write_joint_state_to_sim(joint_pos_all, joint_vel_all, env_ids=env_ids)
```

Comment:
Isn't this a slower operation since you are copying over the data twice and also doing a clone operation. Why not keep it cleaner and just operate over the `asset_cfg.joint_indices` in the whole thing?

Something like:

```python
    # extract the used quantities (to enable type-hinting)
    asset: Articulation = env.scene[asset_cfg.name]
    command_term: PhaseCommand = env.command_manager.get_term(command_name)

    # broadcast env_ids if needed to allow double indexing
    if env_ids != slice(None) and asset_cfg.joint_ids != slice(None):
        env_ids = env_ids[:, None]

    # read state
    ....

    # clamp joint pos to limits
    joint_pos_limits = asset.data.soft_joint_pos_limits[env_ids, asset_cfg.joint_ids]
    joint_pos = joint_pos.clamp_(joint_pos_limits[..., 0], joint_pos_limits[..., 1])
    # clamp joint vel to limits
    joint_vel_limits = asset.data.soft_joint_vel_limits[env_ids, asset_cfg.joint_ids]
    joint_vel = joint_vel.clamp_(-joint_vel_limits, joint_vel_limits)

    # make sure env ids dimensions are compatible
    if env_ids != slice(None) and asset_cfg.joint_ids != slice(None):
        env_ids = env_ids.squeeze(1)
    # set into the physics simulation
    asset.write_joint_state_to_sim(joint_pos, joint_vel, env_ids=env_ids, joint_ids=asset_cfg.joint_ids)

```